confbot_description/urdf/confbot.urdf

```diff
@@ -32,33 +57,184 @@
+    </collision>
     <visual>
       <origin rpy="1.57 0 0" xyz="0 0 0"/>
       <geometry>
-        <cylinder radius="0.3" length="0.2"/>
+        <cylinder radius="0.1" length="0.05"/>
+      </geometry>
+      <material name="white"/>
+    </visual>
+  </link>
+  <joint name="caster_wheel_front_joint" type="continuous">
+    <origin rpy="0 0 0" xyz="0.4 0.0 -0.05"/>
+    <axis xyz="1.0 1.0 0"/>
+    <parent link="base_link"/>
+    <child link="caster_wheel_front_link"/>
+  </joint>
+  <link name="caster_wheel_front_link">
+    <inertial>
+      <mass value="1"/>
+      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
+      <origin rpy="0 0 0" xyz="0 0 0"/>
+    </inertial>
+    <collision>
+      <!--origin rpy="1.57 0 0" xyz="0 0 0"/-->
+      <origin rpy="0 0 0" xyz="0 0 0"/>
+      <geometry>
+        <sphere radius="0.05"/>
+      </geometry>
+      <material name="white"/>
+    </collision>
+    <visual>
+      <origin rpy="0 0 0" xyz="0 0 0"/>
+      <geometry>
+        <sphere radius="0.05"/>
+      </geometry>
+      <material name="white"/>
+    </visual>
+  </link>
+  <joint name="caster_wheel_rear_joint" type="continuous">
+    <origin rpy="0 0 0" xyz="-0.4 0.0 -0.05"/>
+    <axis xyz="1.0 1.0 0"/>
+    <parent link="base_link"/>
+    <child link="caster_wheel_rear_link"/>
+  </joint>
+  <link name="caster_wheel_rear_link">
+    <inertial>
+      <mass value="1"/>
+      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
+      <origin rpy="0 0 0" xyz="0 0 0"/>
+    </inertial>
+    <collision>
+      <!--origin rpy="1.57 0 0" xyz="0 0 0"/-->
+      <origin rpy="0 0 0" xyz="0 0 0"/>
+      <geometry>
+        <sphere radius="0.05"/>
+      </geometry>
+      <material name="white"/>
+    </collision>
+    <visual>
+      <origin rpy="0 0 0" xyz="0 0 0"/>
+      <geometry>
+        <sphere radius="0.05"/>
       </geometry>
       <material name="white"/>
     </visual>
   </link>
+  <gazebo>
+    <plugin name='diff_drive' filename='libgazebo_ros_diff_drive.so'>
+
+      <ros>
+        <namespace>/</namespace>
+        <argument>cmd_vel:=cmd_vel</argument>
+        <argument>odom:=odom</argument>
+      </ros>
+
+      <!-- wheels -->
+      <left_joint>wheel_l_joint</left_joint>
+      <right_joint>wheel_r_joint</right_joint>
+
+      <!-- kinematics -->
+      <wheel_separation>1.2</wheel_separation>
+      <wheel_diameter>0.6</wheel_diameter>
+
+      <!-- limits -->
+      <max_wheel_torque>20</max_wheel_torque>
+      <max_wheel_acceleration>1.0</max_wheel_acceleration>
+
+      <!-- output -->
+      <publish_odom>true</publish_odom>
+      <publish_odom_tf>true</publish_odom_tf>
+      <publish_wheel_tf>true</publish_wheel_tf>
+
+      <odometry_frame>odom</odometry_frame>
+      <robot_base_frame>base_link</robot_base_frame>
+
+    </plugin>
+  </gazebo>
+
+<gazebo reference="laser_frame">
+    <!--sensor type="gpu_ray" name="lidar"-->
+    <sensor type="ray" name="lidar">
```

Comment:
I'm using ray here because `gpu_ray` returns infinity for all points even the one hitting obstacles.
also it prints this warning:
```
[gazebo-1] [Wrn] [msgs.cc:1852] Conversion of sensor type[gpu_ray] not supported.
```

Reply:
That warning can be ignored.

Does your model have visuals besides collisions? The `gpu_ray` sees visuals, while the `ray` sees collisions.

Can you reproduce the issue with this demo world? https://github.com/ros-simulation/gazebo_ros_pkgs/blob/ros2/gazebo_plugins/worlds/gazebo_ros_ray_sensor_demo.world

Reply:
I'm using only primitive shapes that I dropped in the world from the Gazebo UI, I think they have both visual and collision.

Thanks for the pointer, we'll try it on the demo world to see if it's the same.

Note: I'm compiling the gazebo plugins from source without any specific flags, maybe GPU support is not enabled?

Reply:
>  I think they have both visual and collision.

Yup, they do

> I'm compiling the gazebo plugins from source without any specific flags, maybe GPU support is not enabled?

There are no flags to disable GPU support, so compilation should be working correctly. At runtime, if `gzclient` works, then I'd expect the rendering sensors to work as well. You could double-check by dropping a camera in the world and seeing if the images show the shapes.

Reply:
@chapulina Sorry for the delay, I fInally got a chance to try the demo world (using the dashing debs), and I'm facing the same issue:

command ran:
```
gazebo --verbose /opt/ros/dashing/share/gazebo_plugins/worlds/gazebo_ros_ray_sensor_demo.world```
```

<details><summary>Regular ray sensor works (some non inf values)</summary>

```
$ ros2 topic echo /ray/laserscan
header:
  stamp:
    sec: 13
    nanosec: 25000000
  frame_id: ray_link
angle_min: -0.5235999822616577
angle_max: 0.5235999822616577
angle_increment: 0.0035023412201553583
time_increment: 0.0
scan_time: 0.0
range_min: 0.05000000074505806
range_max: 50.0
ranges: [.inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 2.0448575019836426, 2.0313127040863037, 2.0236432552337646, 2.018503189086914, 2.0150201320648193, 2.0128238201141357, 2.0117366313934326, 2.0116827487945557, 2.0126583576202393, 2.0147311687469482, 2.0180652141571045, 2.0230026245117188, 2.030329465866089, 2.0427944660186768, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 1.455374836921692, 1.45494544506073, 1.4545341730117798, '...']
intensities: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 100.0, 100.0, '...']



```

</details>

<details><summary>gpu_ray sensor returns only infinity</summary>

```
$ ros2 topic echo /gpu_ray/laserscan
header:
  stamp:
    sec: 7
    nanosec: 0
  frame_id: ray_link
angle_min: -0.5235999822616577
angle_max: 0.5235999822616577
angle_increment: 0.0035023412201553583
time_increment: 0.0
scan_time: 0.0
range_min: 0.05000000074505806
range_max: 50.0
ranges: [-.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, -.inf, '...']
intensities: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, '...']

```

</details>


I tried the camera demo and can see the objects successfully:
```
gazebo --verbose  /opt/ros/dashing/share/gazebo_plugins/worlds/gazebo_ros_camera_demo.world
```

Reply:
Interesting, I get very similar readings from both of them, I'm on a source install.

```
$ ros2 topic echo /gpu_ray/laserscan
header:
  stamp:
    sec: 139
    nanosec: 2000000
  frame_id: ray_link
angle_min: -0.5235999822616577
angle_max: 0.5235999822616577
angle_increment: 0.0035023412201553583
time_increment: 0.0
scan_time: 0.0
range_min: 0.05000000074505806
range_max: 50.0
ranges: [.inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 2.0503170490264893, 2.033238172531128, 2.024780511856079, 2.0192136764526367, 2.0154314041137695, 2.012951612472534, 2.011847734451294, 2.0118982791900635, 2.0129361152648926, 2.0150675773620605, 2.0186455249786377, 2.0241785049438477, 2.032437562942505, 2.0490212440490723, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 1.454956293106079, 1.4545066356658936, '...']
intensities: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, '...']
---
```

```
$ ros2 topic echo /ray/laserscan
header:
  stamp:
    sec: 153
    nanosec: 18000000
  frame_id: ray_link
angle_min: -0.5235999822616577
angle_max: 0.5235999822616577
angle_increment: 0.0035023412201553583
time_increment: 0.0
scan_time: 0.0
range_min: 0.05000000074505806
range_max: 50.0
ranges: [.inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 2.0448575019836426, 2.0313127040863037, 2.0236432552337646, 2.018503189086914, 2.0150201320648193, 2.0128238201141357, 2.0117366313934326, 2.0116827487945557, 2.0126583576202393, 2.0147311687469482, 2.0180652141571045, 2.0230026245117188, 2.030329465866089, 2.0427944660186768, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, .inf, 1.455374836921692, 1.45494544506073, 1.4545341730117798, '...']
intensities: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 60.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 100.0, 100.0, 100.0, '...']
---
```

Do the GPU readings look right on Gazebo's topic viewer (`Ctrl + T`, and you need to zoom in):

![topicviewer](https://user-images.githubusercontent.com/5751272/58994650-1cdbd280-87a6-11e9-9de3-8c731461599c.png)

I can only think it's an issue with your graphics, but I'm not sure why the RGB camera works. Are the point cloud messages working?

Reply:
> I can only think it's an issue with your graphics, but I'm not sure why the RGB camera works. 

Yeah I had some issues of RViz being very slow as well. I ended up reinstalling all the graphics drivers and it seems to have fixed the issue !

Thanks for taking the time to reproduce :bowing_man: !